src/component/suggestions/index.jsx

```diff
@@ -1,4 +1,42 @@
+import React, { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { useNavigate } from "react-router-dom";
+import { authState, userState, getAllUser, followUnfollowUser } from 'features';
+
 export const Suggestions = () => {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const { user, token } = useSelector(authState);
+  const { allUsers } = useSelector(userState);
+
+  const [suggestions, setSuggestion] = useState([]);
+  
+  useEffect(() => {
+    setSuggestion(allUsers
+      .filter(u => u.username !== user.username)
+      .filter(item => !user.following?.find(followingUser => followingUser._id === item._id))
+    );
+  },
+    [user, allUsers]);
+
+  const isFollowing = followUserId => user.followoing?.find(u => u._id === followUserId) ? true : false;
+
+  const followUserHandler = async followUserId => {
+    try {
+      const response = await dispatch(followUnfollowUser({ 
+        token: token, 
```

Comment:
suggestion: try [axios default headers](https://axios-http.com/docs/config_defaults), so no need to pass the header again and again for each network call.